firefox-ios/Client/Frontend/Browser/StartAtHome/StartAtHomeMiddleware.swift

```diff
@@ -0,0 +1,116 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Redux
+import Shared
+
+final class StartAtHomeMiddleware {
+    private let windowManager: WindowManager
+    private let logger: Logger
+    private let prefs: Prefs
+    private let dateProvider: DateProvider
+
+    init(profile: Profile = AppContainer.shared.resolve(),
+         windowManager: WindowManager = AppContainer.shared.resolve(),
+         logger: Logger = DefaultLogger.shared,
+         dateProvider: DateProvider = SystemDateProvider()) {
+        self.windowManager = windowManager
+        self.logger = logger
+        self.prefs = profile.prefs
+        self.dateProvider = dateProvider
+    }
+
+    lazy var startAtHomeProvider: Middleware<AppState> = { state, action in
+        switch action.actionType {
+        case StartAtHomeActionType.didBrowserBecomeActive:
+            let shouldStartAtHome = self.startAtHomeCheck(windowUUID: action.windowUUID)
+            store.dispatch(
+                StartAtHomeAction(
+                    shouldStartAtHome: shouldStartAtHome,
+                    windowUUID: action.windowUUID,
+                    actionType: StartAtHomeMiddlewareActionType.startAtHomeCheckCompleted
+                )
+            )
+        default: break
+        }
+    }
+
+    private func tabManager(for uuid: WindowUUID) -> TabManager {
+        guard uuid != .unavailable else {
+            assertionFailure()
+            logger.log("Unexpected or unavailable window UUID for requested TabManager.", level: .fatal, category: .tabs)
+            return windowManager.allWindowTabManagers().first!
+        }
+
+        return windowManager.tabManager(for: uuid)
+    }
+
+    /// Checks whether the app should start at the homepage and initiates the process if applicable.
+    /// This method uses `StartAtHomeHelper` to determine if the app should launch with a homepage tab,
+    /// based on user preferences and session state.
+    ///
+    /// - Returns: `true` if a homepage tab was selected and displayed, `false` otherwise.
+    private func startAtHomeCheck(windowUUID: WindowUUID) -> Bool {
```

Comment:
kept most of the logic as existing, but add `tabManager` when needed and added a date provider to inject the date which makes the code more testable